### ISSUE TITLE
Midpoint Review - Accessibility Feedback - Disability Benefits Experience Team 1, 21-526EZ, Toxic Exposure Page

### ISSUE THREAD

**Issue by briandeconinck** [department-of-veterans-affairs/va.gov-team#61268]
## VFS actions
- [x] Assign this ticket to the team member(s) responsible for addressing feedback provided by Platform.
- [x] Comment on this ticket:
  - [ ] If the Platform reviewer has any **Thoughts/Questions** that require responses.
  - [ ] When **Must** feedback has been incorporated. As appropriate, link to any other GitHub issues or PRs related to this feedback.
  - [ ] When **Should/Consider** feedback has been incorporated, or if any feedback will not be addressed. As appropriate, link to any other GitHub issues or PRs related to this feedback.
- [x] Close the ticket when all feedback has been addressed.

## Thoughts/questions

-

## Feedback
Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

- **Should:** On the [evidence overview page](https://www.sketch.com/s/c353146f-eceb-4626-9918-7603c693417f/a/4o5kl0J), work with content to add some kind of heading structure to whatever extent that's possible. That will help sighted users scan content and screen reader users navigate the content.
- **Should:** When you have explanatory notes after fields (eg. on the initial [exposure question](https://www.sketch.com/s/c353146f-eceb-4626-9918-7603c693417f/a/lRGjvPe)), try to move them either above the question label or between the label and the inputs --- to whatever extent that's possible with the content. The main concern is that the note may impact how a user answers the question, and placing it after the input means some users might not encounter the text when they need it. Ideally, if the content supports it, I would like to see it implemented as [hint text](https://design.va.gov/components/form/radio-button#hint-text).
   - **Engineering note:** If hint text isn't practical here, then when coded you'll need to make sure that text gets programmatically associated to the question some other way. For a fieldset, something like this is likely your best bet for compatibility across screen readers:
   ```
   <fieldset aria-labelledby="legend-id note-id">
      <legend id="legend-id">Question text?</legend>
      [radio buttons or checkboxes]
      <p id="note-id">Note: Some helpful explanatory text</p>
   </fieldset>
   ```
- **Consider:** I did a poor job explaining this one in the meeting, so hopefully I can do a better job in writing! When you ask for [exposure dates](https://www.sketch.com/s/c353146f-eceb-4626-9918-7603c693417f/a/DEwygzk), it looks like you're using the [list and loop pattern](https://design.va.gov/patterns/ask-users-for/multiple-responses#how-to-design-and-build---single-page) where a user enters a date, then there's a button to add more dates. Typically when you click that button, the previous entry collapses into a card and you get a fresh set of inputs.
   
   That pattern isn't great for screen reader users. It's not terrible either when it's short limited data like this. But it's a little tricky for screen reader users to navigate back to a previous item to edit it if they realize they made a mistake.
   
   I would recommend looking at some of the other ways this has been implemented to see if they might be a good fit, like the [sub-task pattern](https://design.va.gov/patterns/help-users-to/complete-a-sub-task) or some of the [variations on the existing list and loop pattern](https://design.va.gov/patterns/ask-users-for/multiple-responses#how-to-design-and-build---add-item). They may not fit the use case, and/or your engineers might not have the bandwidth to implement. But something to experiment with either now or for a future iteration.

**Comment by lydiahooper** [comment]
@christinec-fftc for your awareness - please feel free to add any of your input to this ticket or whenever/wherever relevant.

**Comment by lydiahooper** [comment]
We are waiting for CAIA review (see epic for link to that ticket), and have notified them there are relevant items in this ticket.

**Comment by shannonkford** [comment]
**Should**: On the evidence overview page, work with content to add some kind of heading structure to whatever extent that's possible. That will help sighted users scan content and screen reader users navigate the content.

**Response**
Changes to this page are now out of scope; will take up your feedback when we return to this content.

**Should**: When you have explanatory notes after fields (eg. on the initial exposure question), try to move them either above the question label or between the label and the inputs --- to whatever extent that's possible with the content. The main concern is that the note may impact how a user answers the question, and placing it after the input means some users might not encounter the text when they need it. Ideally, if the content supports it, I would like to see it implemented as hint text.

Engineering note: If hint text isn't practical here, then when coded you'll need to make sure that text gets programmatically associated to the question some other way. For a fieldset, something like this is likely your best bet for compatibility across screen readers:
<fieldset aria-labelledby="legend-id note-id">
   <legend id="legend-id">Question text?</legend>
   [radio buttons or checkboxes]
   <p id="note-id">Note: Some helpful explanatory text</p>
</fieldset>

**Response**
Changes have been made to this screen that remove the note and add hint text.

**Consider:** I did a poor job explaining this one in the meeting, so hopefully I can do a better job in writing! When you ask for exposure dates, it looks like you're using the list and loop pattern where a user enters a date, then there's a button to add more dates. Typically when you click that button, the previous entry collapses into a card and you get a fresh set of inputs.

That pattern isn't great for screen reader users. It's not terrible either when it's short limited data like this. But it's a little tricky for screen reader users to navigate back to a previous item to edit it if they realize they made a mistake.

I would recommend looking at some of the other ways this has been implemented to see if they might be a good fit, like the sub-task pattern or some of the variations on the existing list and loop pattern. They may not fit the use case, and/or your engineers might not have the bandwidth to implement. But something to experiment with either now or for a future iteration.

**Response**
Noted! We will do some checking, but may need to push this to a later date as there are multiple instances of this pattern throughout the form.
